### PR TITLE
Allow users to set tokens for connections to ArcGIS Feature/Map servers

### PR DESCRIPTION
**Pull request by nyalldawson**
Without this it can be impossible to connect to authenticated/private servers.

Tokens are set by creating an "ESRI token based authentication" method from the standard QGIS Authentication settings and associating with the connection.

### CODE REVIEW

**Comment by nyalldawson** [comment]
@elpaso how's this look to you?

File: src/providers/arcgisrest/qgsafsdataitems.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ QVector<QgsDataItem *> QgsAfsRootItem::createChildren()
 {
   QVector<QgsDataItem *> connections;
 
-  const QStringList connectionList = QgsOwsConnection::connectionList( "arcgisfeatureserver" );
+  const QStringList connectionList = QgsOwsConnection::connectionList( "ARCGISFEATURESERVER" );
```
Comment (elpaso) [inline src/providers/arcgisrest/qgsafsdataitems.cpp]: string literal?

File: src/auth/esritoken/qgsauthesritokenmethod.cpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+QString QgsAuthEsriTokenMethod::displayDescription() const
+{
+  return tr( "ESRI token based authentication" );
+}
+
+bool QgsAuthEsriTokenMethod::updateNetworkRequest( QNetworkRequest &request, const QString &authcfg,
+    const QString &dataprovider )
+{
+  Q_UNUSED( dataprovider )
+  QMutexLocker locker( &mMutex );
```
Comment (elpaso) [inline src/auth/esritoken/qgsauthesritokenmethod.cpp]: do you really need the locker here? I see that getMethodConfig also locks

**Review by elpaso — APPROVED** [review]
Look good, just left a couple of comments

**Pull request merged**